client/my-sites/plan-features/header.jsx

```diff
@@ -350,5 +364,6 @@ export default connect( ( state, { isInSignup, planType, relatedMonthlyPlan } )
 		isYearly,
 		relatedYearlyPlan: isYearly ? null : getPlanBySlug( state, getYearlyPlanByMonthly( planType ) ),
 		siteSlug: getSiteSlug( state, selectedSiteId ),
+		countryCode: requestGeoLocation().data || 'US',
```

Comment:
I must say I don't favor side effects in `mapStateToProps`... but others would likely disagree with me (see #24631 for context). 

Essentially, usages like this make me be against approaches like #24631 - it might be a simple HTTP request use case, but when it gets to data that can be useful in multiple locations across the entire app, I think gets different. For example, only this PR introduces 2 distinct usages of this.

That being said, why do we even keep this here? Why not call it in a separate method for example, and cache it in a variable or something?

Finally, if that's your preference, I'm fine to go ahead with it for the purpose of this PR - we already have similar usages in the codebase, so if we take a different approach, we'd need to refactor them as well.

Reply:
I'm just using the same format I've seen used in the examples I've been handled :man_shrugging: 

Since this is a test, and whatever it happens we would either remove this code or refactor it into something definitive, I would rather prefer to keep it like this _by now_

Reply:
Sounds good 👍